src/resolver/resolver.zig

```diff
@@ -3278,8 +3278,7 @@ pub const Resolver = struct {
 
         const str = brk: {
             if (std.fs.path.isAbsolute(sliced.slice())) break :brk sliced.slice();
-            const dir_path_buf = bufs(.node_modules_paths_buf);
-            break :brk r.fs.joinBuf(&[_]string{ r.fs.top_level_dir, sliced.slice() }, dir_path_buf);
+            break :brk bun.path.join(&[_]string{ r.fs.top_level_dir, sliced.slice() }, .auto);
```

Comment:
This will cause strange memory errors if the resolver is run recursively. That is why we were using `bufs` here instead of the implicit default buffer that bun.path.join uses.